editor/import/resource_importer_scene.cpp

```diff
@@ -372,6 +372,194 @@ static void _pre_gen_shape_list(Ref<ImporterMesh> &mesh, Vector<Ref<Shape3D>> &r
 	}
 }
 
+struct ScalableNodeCollection {
+	List<Node3D *> node_3ds;
+	List<Ref<ImporterMesh>> importer_meshes;
+	List<Ref<Skin>> skins;
+	List<Ref<Animation>> animations;
+};
+
+void _rescale_importer_mesh(float p_scale, Ref<ImporterMesh> p_mesh, bool is_shadow = false) {
+	// MESH and SKIN data divide, to compensate for object position multiplying.
+
+	int surf_count = p_mesh->get_surface_count();
+	struct LocalSurfData {
+		Mesh::PrimitiveType prim;
+		Array arr;
+		Array bsarr;
+		Dictionary lods;
+		String name;
+		Ref<Material> mat;
+		int fmt_compress_flags;
+	};
+
+	List<LocalSurfData> surf_data_by_mesh;
+
+	for (int surf_idx = 0; surf_idx < surf_count; surf_idx++) {
+		Mesh::PrimitiveType prim = p_mesh->get_surface_primitive_type(surf_idx);
+		int fmt_compress_flags = p_mesh->get_surface_format(surf_idx);
+		Array arr = p_mesh->get_surface_arrays(surf_idx);
+		String name = p_mesh->get_surface_name(surf_idx);
+		Dictionary lods = Dictionary(); // p_mesh.surface_get_lods(surf_idx) # get_lods(mesh, surf_idx)
+		Ref<Material> mat = p_mesh->get_surface_material(surf_idx);
+		// print("About to multiply mesh vertices by " + str(p_scale) + ": " + str(arr[ArrayMesh.ARRAY_VERTEX][0]))
+
+		Vector<Vector3> vertex_array = arr[ArrayMesh::ARRAY_VERTEX];
+
+		int vert_arr_len = vertex_array.size();
+		int i = 0;
+		while (i < vert_arr_len) {
+			vertex_array.set(i, vertex_array[i] * p_scale);
+			i += 1;
+		}
+		arr[ArrayMesh::ARRAY_VERTEX] = vertex_array;
+
+		Array blendshapes;
+
+		for (int bsidx = 0; bsidx < p_mesh->get_blend_shape_count(); bsidx++) {
+			i = 0;
+
+			Array current_bsarr = p_mesh->get_surface_blend_shape_arrays(surf_idx, bsidx);
+			Vector<Vector3> current_bs_vertex_array = current_bsarr[ArrayMesh::ARRAY_VERTEX];
+
+			int current_bs_vert_arr_len = current_bs_vertex_array.size();
+			i = 0;
+			while (i < current_bs_vert_arr_len) {
+				current_bs_vertex_array.set(i, current_bs_vertex_array[i] * p_scale);
+				i += 1;
+			}
+			current_bsarr[ArrayMesh::ARRAY_VERTEX] = current_bs_vertex_array;
```

Comment:
This code is really messy.

I did things this way in my GDScript implementation because we do not currently have access to a better API for modifying ImporterMesh.

But since we're making a core engine change, it might be better to provide access to the vertex positions of EditorSceneImporterMesh so that they can be modified without deleting and readding everything.

Here's or chance to improve the APIs to make them less clunky.